src/data/actions/courseInfo.js

```diff
@@ -184,16 +178,14 @@ function createCourse(courseData) {
   };
 }
 
-function handleCourseRuns(dispatch, courseRunData, course, submitReview) {
-  // We'll use the presence of the course run status property to signal an internal
-  // review process because it is only present when we pass in a single run for internal
-  // review, rather than an array of runs on a regular edit
-  const internalReview = !!courseRunData.status;
-  const functionCall = internalReview ? ApiFunctionEnum.INTERNAL_REVIEW_FUNCTION :
-    ApiFunctionEnum.EDIT_RUNS_FUNCTION;
+function handleCourseRuns(dispatch, courseRunData, course, submitReview, internalReview) {
   // make course copy so we are not re-assigning properties of this functions original params
   const newCourse = Object.assign({}, course);
-  DiscoveryDataApiService[functionCall](courseRunData).then((runResponse) => {
+  let sendData = DiscoveryDataApiService.editCourseRuns;
+  if (internalReview) {
+    sendData = data => Promise.all([DiscoveryDataApiService.internalReviewEdit(data)]);
+  }
+  sendData(courseRunData).then((runResponse) => {
```

Comment:
I think this whole arrangement is _cleaner_, but not ideal. This layer shouldn't know about details like this. Ideally `handleCourseSubmit` would either pass down this `sendData` callback itself or really, start a different thread than `editCourse` which mostly assumes its dealing with lists.

But... this fix is slightly better and fairly easy. So I did this.